### DISCUSSION TITLE
[Feature] Unlink faces

### DISCUSSION THREAD

**Discussion by go-vegan** [immich-app/immich#5276] [Feature Request]
### The feature

Sometimes the model links between different persons. I found this happening especially with baby faces.
An option to unlink individual pictures from a person would provide the fine-grained manual touch needed.

For clarification: this is not about unmerging merged faces. Hence the term "unlink".

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by davidpan** [comment]
Suggested Features:
- Remove one or more photos from a person's facial recognition
results.
- Associate one or more photos with a person's facial recognition
results.

When removing or associating photos, ensure that the corresponding facial recognition results are also cleared, and avoid any re-categorization by existing models.

The function above is also useful for improving future models and learning local algorithm migration.

**Comment by davidpan** [comment]
A search reveals that the feature is already in development. https://github.com/immich-app/immich/pull/4717

**Reply by martabal** [reply]
Un-assigning faces is not being developed atm. Re-assigning is #4949. I'm waiting for this PR to be merged first before working on it.

**Comment by vonmillhausen** [comment]
I'm here specifically to say I'm looking for the ability to unlink/unassign a face that has already been assigned. Many of the images in my collection feature group shots of sporting events that contain crowds. I've run into the situation where background faces in the crowd look similar to specific sportspeople I've set up as people, and are thus assigned to that person - even though I'm not interested in the background faces at all. Currently there is no way to unassign faces once they've been assigned to a person - you can reassign them to someone else, but that's not useful if you don't want to assign them at all.

So my +1 for the ability to unassign an assigned face.

**Reply by TimPasquini** [reply]
@vonmillhausen You definitely can and it's easy to do once you know how, the UI design just doesn't make it obvious/intuitive on how to do it.  

Open the photo and go into the edit faces mode, then click the pencil next to the face you want unassign. Click the "+" button all the way in the top right corner to convert the assigned face into a "new" unassigned face. This will de-link it from any existing persons and then the face can be safely hidden if it's just a random background face.

**Reply by vonmillhausen** [reply]
@TimPasquini Ahh, thank you - that did indeed work! I had been afraid that would have created a new "nameless" person with one photo, that would have shown up in the Explore view, and that it would have confused the bejeebus out of the face recognition system going forward... but it didn't, and I was able to correct the incorrectly assigned faces. Thanks again!

**Comment by J51corp** [comment]
Added to this as mine was closed as dup. Still would like to see this change. 

The feature
I did a mass import, and the face matching matched a lot of photos to one person, will lots of other individuals. There is a "Fix incorrect match" option, and in that option I can create a new person, or assign to another person, but no option to unmatch them all selected.

Not sure if the logical process would be to resend them to be scanned and re-matched or not.

I also have one image (of a cat) that has captured a lot of random matches as well. (car crash, meme's, eggs, lots of group photos, a bird, and 429 random photos) this would be helpful on this one too.

Thanks,

**Comment by bo0tzz** [comment]
Closing as this is possible these days.
